plugin/sampling/strategystore/adaptive/postaggregator.go

```diff
@@ -244,65 +176,47 @@ func addJitter(jitterAmount time.Duration) time.Duration {
 	return (jitterAmount / 2) + time.Duration(rand.Int63n(int64(jitterAmount/2)))
 }
 
-func (p *Processor) runCalculationLoop() {
```

Comment:
Can we go back and NOT move this code? Just change the receiver:
```
func (p *StrategyStore) runCalculationLoop() {
```
We need to minimize the changes when doing refactoring, it's impossible to ascertain otherwise if the changes are equivalent.

Reply:
Actually, I haven't done any major changes in it, i just extracted this from `runCalculationLoop` to create a helper function `runCalculation`, which is combined with aggregator goroutine.

```go
endTime := time.Now().Add(p.Delay * -1)
	startTime := p.lastCheckedTime
	throughput, err := p.storage.GetThroughput(startTime, endTime)
	if err != nil {
		p.logger.Error(getThroughputErrMsg, zap.Error(err))
		return
	}
	aggregatedThroughput := p.aggregateThroughput(throughput)
	p.prependThroughputBucket(&throughputBucket{
		throughput: aggregatedThroughput,
		interval:   endTime.Sub(startTime),
		endTime:    endTime,
	})
	p.lastCheckedTime = endTime
	// Load the latest throughput so that if this host ever becomes leader, it
	// has the throughput ready in memory. However, only run the actual calculations
	// if this host becomes leader.
	// TODO fill the throughput buffer only when we're leader
	if p.isLeader() {
		startTime := time.Now()
		probabilities, qps := p.calculateProbabilitiesAndQPS()
		p.Lock()
		p.probabilities = probabilities
		p.qps = qps
		p.Unlock()

		p.calculateProbabilitiesLatency.Record(time.Since(startTime))
		p.runBackground(p.saveProbabilitiesAndQPS)

```

Reply:
You have over 150 lines removed from processor.go and as many added to storage.go, so that's a very large (and unnecessary) change. I cannot compare them across files as a diff.

Reply:
ok

Reply:
please review